app/src/main/java/com/example/grocerieswizard/shop/subshop/SubShopAdapter.java

```diff
@@ -92,18 +93,7 @@ public void bind(SubShoppingItem subShoppingItem) {
             binding.subIngredientCheckbox.setOnCheckedChangeListener(null);
             binding.subIngredientCheckbox.setChecked(subShoppingItem.getChecked());
             Log.d(TAG, "bind: setCheckedSubItem: " + subShoppingItem.getRecipeName());
-            binding.subIngredientCheckbox.setOnCheckedChangeListener((buttonView, isChecked) -> {
```

Comment:
Try to set listeners in onCreateViewHolder. onBind function should be as simple as possible, only setting the state, because it is trigerred many times (especially if u have large list). F.e. if you scroll down the onBind function will be called for every item that has been scrolled to, also when you scroll up it will trigger onBind function, so we shouldn't do complicated calculations here.